google/resource_storage_default_object_acl_test.go

```diff
@@ -0,0 +1,213 @@
+					testAccCheckGoogleStorageDefaultObjectAcl(bucketName, roleEntityBasic2),
+					testAccCheckGoogleStorageDefaultObjectAcl(bucketName, roleEntityBasic3_owner),
+				),
+			},
+
+			resource.TestStep{
+				Config: testGoogleStorageDefaultObjectsAclBasic3(bucketName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGoogleStorageDefaultObjectAcl(bucketName, roleEntityBasic2),
+					testAccCheckGoogleStorageDefaultObjectAcl(bucketName, roleEntityBasic3_reader),
+				),
+			},
+
+			resource.TestStep{
+				Config: testGoogleStorageDefaultObjectsAclBasicDelete(bucketName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGoogleStorageDefaultObjectAclDelete(bucketName, roleEntityBasic1),
+					testAccCheckGoogleStorageDefaultObjectAclDelete(bucketName, roleEntityBasic2),
+					testAccCheckGoogleStorageDefaultObjectAclDelete(bucketName, roleEntityBasic3_reader),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckGoogleStorageDefaultObjectAcl(bucket, roleEntityS string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		roleEntity, _ := getRoleEntityPair(roleEntityS)
+		config := testAccProvider.Meta().(*Config)
+
+		res, err := config.clientStorage.DefaultObjectAccessControls.Get(bucket,
+			roleEntity.Entity).Do()
+
+		if err != nil {
+			return fmt.Errorf("Error retrieving contents of storage default Acl for bucket %s: %s", bucket, err)
+		}
+
+		if res.Role != roleEntity.Role {
+			return fmt.Errorf("Error, Role mismatch %s != %s", res.Role, roleEntity.Role)
+		}
+
+		return nil
+	}
+}
+
+func testAccGoogleStorageDefaultObjectAclDestroy(s *terraform.State) error {
+	config := testAccProvider.Meta().(*Config)
+
+	for _, rs := range s.RootModule().Resources {
+
+		if rs.Type != "google_storage_default_object_acl" {
+			continue
+		}
+
+		bucket := rs.Primary.Attributes["bucket"]
+
+		_, err := config.clientStorage.DefaultObjectAccessControls.List(bucket).Do()
+		if err == nil {
+			return fmt.Errorf("Default Storage Object Acl for bucket %s still exists", bucket)
+		}
+	}
+	return nil
+}
+
+func testAccCheckGoogleStorageDefaultObjectAclDelete(bucket, roleEntityS string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		roleEntity, _ := getRoleEntityPair(roleEntityS)
+		config := testAccProvider.Meta().(*Config)
+
+		_, err := config.clientStorage.DefaultObjectAccessControls.Get(bucket, roleEntity.Entity).Do()
+
+		if err != nil {
+			return nil
+		}
+
+		return fmt.Errorf("Error, Object Default Acl Entity still exists %s for bucket %s",
+			roleEntity.Entity, bucket)
+	}
+}
+
+func testGoogleStorageDefaultObjectsAclBasicDelete(bucketName string) string {
+	return fmt.Sprintf(`
+resource "google_storage_bucket" "bucket" {
+	name = "%s"
+}
+
+resource "google_storage_default_object_acl" "acl" {
+	bucket = "${google_storage_bucket.bucket.name}"
+	role_entity = []
+}
+`, bucketName)
+}
+
+func testGoogleStorageDefaultObjectsAclBasic1(bucketName string) string {
```

Comment:
These three tests are exactly the same except for which vars are filled in for the role_entity. How about something like this:
```
func testGoogleStorageDefaultObjectAclBasic(bucketName, roleEntity1, roleEntity2 string) string {
    return fmt.Sprintf(`
// config here
`, bucketName, roleEntity1, roleEntity2)
```